packages/gatsby/src/redux/reducers/queries.ts

```diff
@@ -305,3 +319,14 @@ function registerComponent(
   }
   return component
 }
+
+function trackDirtyQuery(
+  state: IGatsbyState["queries"],
+  queryId: QueryId
+): IGatsbyState["queries"] {
+  if (process.env.GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND) {
+    state.dirtyQueriesListToEmitViaWebsocket.push(queryId)
```

Comment:
Is it OK to have it as Array not Set? I.e. can there be some edge cases with duplicates?

Reply:
I went with array because Set is not serializable and we would have to convert Set to array when we emit it through websocket. There is potential for duplicates, but I think duplicates are not breaking just bit more unnecessary work